go/parquet/internal/utils/bit_packing_avx2_amd64.go

```diff
@@ -45,6 +45,10 @@ func unpack32Avx2(in io.Reader, out []uint32, nbits int) int {
 	buffer.Grow(n)
 	io.CopyN(buffer, in, int64(n))
 
+	if buffer.Len() == 0 {
+		return 0
+	}
```

Comment:
Would it make more sense to shift this above line 42 and have it be:

```go
if n == 0 {
    return 0
}
```

instead?

Reply:
With the dataset I have it makes sense.
I don't have enough experience with arrow to know if `n!=0 && buffer.Len()==0` is possible. In this case, the program would crash.

Reply:
given the way the code is written, `buffer.Len()` should always end up being equal to `n` because we call `buffer.Reset()` and `buffer.Grow(n)` so I think we should make this change.

Reply:
Yes you are right. I moved the condition directly on `nbits`. Since `batch` is already checked, it covers all cases where `n` is equal to 0. Is it OK for you?

Reply:
Because we're dealing with integer division, that doesn't cover all cases. To cover all cases you'd need something like `if batch * nbits < 8` since integer division would result in `n == 0` for any scenario where `batch * nbits` is less than 8 (since that would result in 0 when we divide by 8). So it might just be better to just check if `n == 0` rather than try to find all the cases

Reply:
Of course, I moved the condition on `n`.